pkg/v1/providers/ytt/02_addons/public-cloud-csi/add_aws_csi_driver.yaml

```diff
@@ -0,0 +1,24 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
+#@ load("/lib/helpers.star", "ValuesFormatStr")
+#@ load("ebs/aws-ebs-csi-driver_data.lib.yaml", "ebsdatavalues")
+
+#@ if data.values.PROVIDER_TYPE == "aws" and not data.values.IS_WINDOWS_WORKLOAD_CLUSTER:
```

Comment:
I think we need to gate creation of these secrets  with the function `enable_csi_driver` so that we enable only for tkr>=1.23.0

Reply:
We have some discussion here, so the idea is to have csi driver installed for **all** three tkr versions installed by tanzu 1.6, but set default storage class differently, for 1.21 and 1.22, default sc will be in-tree driver, for 1.23, default sc will be  csi driver, this aligns with Kubernetes CSI's deprecation strategy (for 1.21 and 1.22, in-tree is still the default driver but from 1.23, default one is csi driver);
Also have csi driver installed in 1.21 and 1.22 are functional well, and easier for later workload cluster upgrade.

hope this makes sense.